### DISCUSSION TITLE
RB5009 + ODI DFP-34X-2C2 slower than ISP's ONU

### DISCUSSION THREAD

**Discussion by SionFE** [Anime4000/RTL960x#245] [Q&A] [answered]
Hello. 
I replaced ZTE F670 ONU, provided by ISP, with Mikrotik RB5009UG+S+IN and ODI DFP-34X-2C2. ROS 7.13.3 and SFP FW is 220923. ISP provides me a 1 Gbit plan.
Did some comparison using Speedtest. Here are the results:
ZTE F670 - stable 950 download, 940 upload.
RB5009 + ODI - 940 download with periodical throttling to 890, stable 940 upload.

LAN_SDS_MODE set to 6. On Mikrotik 2500BaseX set. What could be the possible reason for such behaviour?

**Comment by Anime4000** [comment]
Mikrotik flow control between PON Stick at this moment kind of bugged, try this:

Switch > Port
![image](https://github.com/Anime4000/RTL960x/assets/1908715/6a92440c-244d-463b-b53c-e452849067ef)

**Reply by SionFE** [reply]
Tried this, the results became worse:
No more than 110 Mbit/s on download, same 940 Mbit/s on upload.

**Reply by Anime4000** [reply]
have you tried
`flash set OMCI_TM_OPT 0` ?

This look same like issue #35

**Reply by SionFE** [reply]
Yes, tried right now. This is the result with `OMCI_TM_OPT 0` and switch port settings enabled:
![1](https://github.com/Anime4000/RTL960x/assets/90137239/5d813f1a-a4fd-41b9-9e4d-c88d69449176)

And this is the result with `OMCI_TM_OPT 0` and switch port settings disabled (this is when throttling occurs):
![image](https://github.com/Anime4000/RTL960x/assets/90137239/1ba437f2-a9bb-4b09-8c84-51d9eaa5680c)

When there is no throttling I get 940/940 without switch port settings.

**Reply by Anime4000** [reply, marked as answer]
since you doing `2.5G baseX` in Mikrotik, try this via PON Telnet:
```
/bin/diag bandwidth set egress port all rate 4194296
/bin/diag bandwidth set ingress port all rate 4194296
```

assume you have access PON IP Address via NAT

that command wont survive across reboot, apply when you have Internet

**Reply by SionFE** [reply]
Look like this helped me. Thank you.

**Comment by Gradius2** [comment]
Those cmds are for Mikrotik only, right?

/bin/diag port set auto-nego port all ability asy-flow-control
/bin/diag bandwidth set egress port all rate 4194296
/bin/diag bandwidth set ingress port all rate 4194296

I have a media converter, and just using OMCI_TM_OPT 0, won't fix it.

What about of us, with just a media converter ?
